android/src/com/frostwire/android/core/FileDescriptor.java

```diff
@@ -82,6 +84,39 @@ public FileDescriptor(int id, String artist, String title, String album, String
         ensureCorrectMimeType(this);
     }
 
+    public FileDescriptor(Bundle bundle) {
+        this(bundle.getInt("id"),
```

Comment:
you mean something like this?

```java
public static FileDescriptor fromBundle(Bundle bundle)
```

I didn't do it like that because my `toBundle()` method is not static, it's simply a way to get a bundle representation from `this`

I could make it something like this and make it more consistent, but not sure if it's better to have less `static` symbols loaded as possible

```
java
public static Bundle toBundle(FileDescriptor fd)
```